src/apps/vpn/connectionhealth.cpp

```diff
@@ -149,7 +149,11 @@ void ConnectionHealth::setStability(ConnectionStability stability) {
     emit GleanDeprecated::instance()->recordGleanEvent(
         GleanSample::connectionHealthNoSignal);
   } else {
-    mozilla::glean::sample::connection_health_stable.record();
+#if defined(MZ_ANDROID) || defined(MZ_IOS)
+    // Count successful health checks only on mobile apps, as they
+    // only do health checks when foregrounded.
```

Comment:
I'm unclear on this point. Why is it only interesting to count connection_health_stable events on mobile?

The code looks right, just don't understand why we don't want this info for desktop.

Reply:
Another point. If this is mobile only it should be collected in daemon, no?

Reply:
Health checks are run every second. The event version of this metric was added under a different understanding, so it's pretty important to get rid of the event version before it gets to a production version of the app.

This PR changes the approach from "count number of events" to "use counter metric", though it seems like your concerns are mostly around the general idea of a "positive health checks" metric, which has been in the tech spec doc for the last 6.5 weeks, and not discussed much there. (Prior to that, it was in the metrics spreadsheet.) I'm responding here to your concerns about the metric, but if the discussion is going to continue, I'd ask that you move it to the tech spec.

Responding to the specific pieces -

**Why we have it** See the "reason we have this" in my comment above, which excerpts from the tech spec. Negative health events are fairly meaningless from the iOS and Android app, as the health checks only run when the app is open. This was an approach that Santiago agreed on early in this project.

**How we plan to analyze it** This one is certainly more than an art than a science, and Santiago is probably better equipped to answer this one. Since we don't get good mobile app data for this, looking at the "bad-health-checks-to-total-health-checks" ratio, along with the absolute number of bad health checks, along with looking at
this in comparison to desktop apps and in comparison to Android daemon reports... that's the sort of things that I'd imagine will be looked at.

**Being implemented ahead of the tech spec being finalized** In a meeting about a month ago, I was encouraged to start implementing the app session metrics and pings of the document, because they was no longer being debated and had been agreed upon. I held off on the pings and ID-related metrics, because they seemed very connected to the daemon discussions, but I started adding many of the app sessino metrics. Given the lack of discussion or changes on these metrics over the last month, this doesn't seem like the group made a bad decision.

**Validation**: This will be handled as part of the [existing validation ticket](https://mozilla-hub.atlassian.net/browse/VPN-4578).